modules/jupyter-widget/src/widget.js

```diff
@@ -53,35 +55,49 @@ export class DeckGLModel extends DOMWidgetModel {
 export class DeckGLView extends DOMWidgetView {
   render() {
     super.render();
-    this.model.on('change:json_input', this.valueChanged, this);
-    loadCss(MAPBOX_CSS_URL);
+    this.model.on('change:json_input', this.valueChanged.bind(this), this);
+    const initialized = this.model.get('initialized');
 
     const containerDiv = document.createElement('div');
 
-    containerDiv.style.height = `${this.model.get('height')}px`;
-    containerDiv.style.width = `${this.model.get('width')}px`;
-    this.el.appendChild(containerDiv);
+    if (!initialized) {
+      containerDiv.style.height = `${this.model.get('height')}px`;
+      containerDiv.style.width = `${this.model.get('width')}px`;
+      this.el.appendChild(containerDiv);
 
-    this.deck = initDeck({
-      mapboxApiKey: this.model.get('mapbox_key'),
-      container: containerDiv,
-      jsonInput: JSON.parse(this.model.get('json_input'))
-    });
+      loadCss(MAPBOX_CSS_URL);
+      initDeck(
+        {
+          mapboxApiKey: this.model.get('mapbox_key'),
+          container: containerDiv,
+          jsonInput: JSON.parse(this.model.get('json_input'))
+        },
+        this,
+        this.handleClick.bind(this)
+      );
+      this.model.set('initialized', true);
+    }
   }
 
   valueChanged() {
-    updateDeck(this.model.get('json_input'), this.deck);
```

Comment:
This invocation looks confusing at it passes in all of deck and not at all clear what is being used. 

I'd expand the arguments and pass them in one by one, or make `updateDeck` a method on `this.deck`.